qiskit/circuit/quantumcircuit.py

```diff
@@ -1533,6 +1534,381 @@ def _rebind_definition(self, instruction, parameter, value):
+                qubits.extend([qarg[j] for j in range(qarg.size)])
+            elif isinstance(qarg, list):
+                qubits.extend(qarg)
+            elif isinstance(qarg, range):
+                qubits.extend(list(qarg))
+            elif isinstance(qarg, slice):
+                qubits.extend(self.qubits[qarg])
+            else:
+                qubits.append(qarg)
+
+        return self.append(Barrier(len(qubits)), qubits, [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def h(self, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.HGate`."""
+        from .library.standard_gates.h import HGate
+        return self.append(HGate(), [qubit], [])
+
+    @deprecate_arguments({'ctl': 'control_qubit', 'tgt': 'target_qubit'})
+    def ch(self, control_qubit, target_qubit,  # pylint: disable=invalid-name
+           *, label=None, ctrl_state=None, ctl=None, tgt=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.CHGate`."""
+        from .library.standard_gates.h import CHGate
+        return self.append(CHGate(label=label, ctrl_state=ctrl_state),
+                           [control_qubit, target_qubit], [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def i(self, qubit, *, q=None):  # pylint: disable=unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.IGate`."""
+        from .library.standard_gates.i import IGate
+        return self.append(IGate(), [qubit], [])
+
+    @deprecate_arguments({'q': 'qubit'})
+    def id(self, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.IGate`."""
+        return self.i(qubit)
+
+    @deprecate_arguments({'q': 'qubit'})
+    def iden(self, qubit, *, q=None):  # pylint: disable=unused-argument
+        """Deprecated identity gate."""
+        warnings.warn('The QuantumCircuit.iden() method is deprecated as of 0.14.0, and '
+                      'will be removed no earlier than 3 months after that release date. '
+                      'You should use the QuantumCircuit.i() method instead.',
+                      DeprecationWarning, stacklevel=2)
+        return self.i(qubit)
+
+    def ms(self, theta, qubits):  # pylint: disable=invalid-name
+        """Apply :class:`~qiskit.circuit.library.standard_gates.MSGate`."""
+        from .library.standard_gates.ms import MSGate
+        return self.append(MSGate(len(qubits), theta), qubits)
+
+    @deprecate_arguments({'q': 'qubit'})
+    def r(self, theta, phi, qubit, *, q=None):  # pylint: disable=invalid-name,unused-argument
+        """Apply :class:`~qiskit.circuit.library.standard_gates.RGate`."""
+        from .library.standard_gates.r import RGate
+        return self.append(RGate(theta, phi), [qubit], [])
+
+    def rccx(self, control_qubit1, control_qubit2, target_qubit):
```

Comment:
always been annoying to me that we use relative and rotation with an r

Reply:
we need to pick one

Reply:
yeah for this one I think we want to allow different synthesis options. relative phase falls under `up_to_diagonal`. in general:

```
synthesis(circ_instance, up_to_phase=True, up_to_diagonal=True, approximation=0.99)
```